caikit_nlp/data_model/text.py

```diff
@@ -25,3 +28,21 @@ class Span(DataObjectBase):
     start: int
     end: int
     text: str
+
+
+@caikit.core.dataobject(package="caikit_data_model.caikit_nlp")
+class Token(DataObjectBase):
+    """Tokens here are the basic units of text. Tokens can be characters, words,
+    sub-words, or other segments of text or code, depending on the method of
+    tokenization chosen or the task being implemented.
+    """
+
+    # NOTE: This data model is purposefully created separate from "Span"
+    # to provide flexibility for adding more token level constructs in future
+    # if needed, like lemma, part of speech etc.
+    span: Span
```

Comment:
as discussed we may want to just flatten this to just `Token` with start, end, etc.